### ISSUE TITLE
Unexpected $ref warning when a definition is declared and used once

### ISSUE THREAD

**Issue by martinsirbe** [IBM/openapi-validator#125]
Validator v0.17.0 with the following v3 OpenAPI specification:
```yaml
openapi: 3.0.0
info:
  description: Pancakes are the best!
  version: 1.33.7
  title: Pancakes API
paths:
  /pancakes:
    get:
      operationId: list_pancakes
      summary: Returns all pancakes.
      description: For pancakes...
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  name:
                    description: The name of the pancake.
                    type: string
```

I see the following warning:
```bash
warnings

  Message :   Response schemas should be defined with a named ref.
  Path    :   paths./pancakes.get.responses.200.content.application/json.schema
  Line    :   17

statistics

  Total number of errors   : 0
  Total number of warnings : 1

  warnings
   1 (100%) : Response schemas should be defined with a named ref.
```

I'm trying to understand whether this is a valid warning, as based on the ["Using $ref"](https://swagger.io/docs/specification/using-ref/) documentation:

> it is common to have some features which you use across several of API resources. In that case, you can create a snippet for such elements in order to use them multiple times when you need it.

it seems that `$ref` was introduced to follow DRY principle, however in this case I can't see the validity of this being a warning as the definition is declared and used only once.

Probably, the desired behaviour would be to check for duplicated definitions before raising the named ref warning.

**Comment by dpopp07** [comment]
At IBM, we use OpenAPI definitions to generate SDKs. One of our primary use-cases for this tool is to ensure that an API definition is fit for SDK Generation (the higher quality the API definition, the higher quality the generated SDK).

This warning you've run into is one that we've added specifically for that purpose. Using a `$ref` for a response provides us with a meaningful name we can use for a response model in the generated SDK. Without it, the name is automatically generated and is less-than-ideal for the UX of the SDK.

Since you are not likely generating SDKs, then I agree - in this scenario you don't really need to use a named reference. To remove the warning, you can turn off [this rule](https://github.com/IBM/openapi-validator#responses-2) in your configuration file.

Let me know if you have any other questions!

**Comment by martinsirbe** [comment]
Thanks @dpopp07 for detailed clarification, this totally makes sense to me. It would be great if this could be documented, to mention that the usage of `$ref` is to achieve a better data model naming when generating SDKs instead of definition reusability.

**Comment by dpopp07** [comment]
That's a fair request, I will update the README to reflect this.

**Comment by jorge-ibm** [comment]
@dpopp07 ^^
